### DISCUSSION TITLE
객체의 프로퍼티 접근 방법에는 어떤 것들이 있나요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#64] [원시 값과 객체의 비교]
- 객체의 프로퍼티 접근 방법에는 어떤 것들이 있나요?
  - 점 표기법과 대괄호 표기법은 각각 어떤 상황에서 사용하나요?

**Comment by pipisebastian** [comment]
> 객체의 프로퍼티 접근 방법에는 어떤 것들이 있나요
- 대괄호 표기법([])과 점 표기법(.)이 있습니다.

> 점 표기법과 대괄호 표기법은 각각 어떤 상황에서 사용하나요?
- 점 표기법의 경우, 변수 네이밍 규칙을 따를 경우, 예를 들어 숫자로 시작하지 않고, 문자, _ 등으로 시작되는 경우에만 사용 가능합니다.
- 대괄호 표기법의 경우, 속성을 문자열로 접근하기 때문에, 변수 네이밍 규칙을 지키지 않더라도 사용가능합니다.

**Comment by publdaze** [comment]
> - 객체의 프로퍼티 접근 방법에는 어떤 것들이 있나요?
- 점 표기법과 대괄호 표기법이 있음
> - 점 표기법과 대괄호 표기법은 각각 어떤 상황에서 사용하나요?
- 점 표기법은 프로퍼티로 식별자 네이밍 규칙과 동일한 규칙을 따름, 따라서 해당 규칙에 부합하면서 간결하게 작성하고 싶을 때 사용하게 되고,
- 프로퍼티 키에 띄어쓰기가 포함되거나, 숫자로 시작되는 등 식별자 네이밍 규칙에 부합하지 않을 경우 대괄호 표기법을 사용할 수 있음
